recorder/src/main/cpp/circular_queue.cc

```diff
@@ -14,9 +14,11 @@ bool CircularQueue::push(const JVMPI_CallTrace &item, ThreadBucket *info) {
         // TODO: have someone review the memory ordering constraints
     } while (!input.compare_exchange_strong(currentInput, nextInput, std::memory_order_relaxed));
```

Comment:
Because push is called from the signal handler, which can be called concurrently on more than 1 thread receiving sigprof close to the same time. We need compare to make sure no 2 attempts take the same slot.